### PR TITLE
feat: add docker build for arm

### PR DESCRIPTION
**Pull request by pradel**
This pull requests add ARM builds to docker so one can host the explorer on an ARM server. The approach is to only do amd build on prs to not increase the build time and build both arm + amd on the main branch for the release.

Related: I added ARM support to the stacks-blockchain-api in this pr https://github.com/hirosystems/stacks-blockchain-api/pull/1947

### CODE REVIEW

**Review by CharlieC3 — APPROVED** [review]
LGTM! Tested the [CI run here](https://github.com/hirosystems/explorer/actions/runs/8972043339/job/24639158373).

**Comment by 314159265359879** [comment]
I would love to see this merged.

**Comment by pradel** [comment]
Is there anything missing from this pr? Would love to run the API on my ARM server :)

**Comment by CharlieC3** [comment]
Sorry for the delay, I'd like to just get sign-off from a repo maintainer before merging. @BLuEScioN or @He1DAr can you review?

**Review by He1DAr — APPROVED** [review]
LGTM

**Pull request merged**
